File: binance4j-nft/src/test/java/com/binance4j/nft/client/NFTClientTest.java
package com.binance4j.nft.client;

import org.junit.jupiter.api.Test;

import com.binance4j.core.exception.ApiException;
import com.binance4j.core.param.FramedPaging;
import com.binance4j.core.param.Paging;
import com.binance4j.core.test.CustomTest;
import com.binance4j.nft.dto.OrderType;
import com.binance4j.nft.param.TransactionHistoryParams;

public class NFTClientTest extends CustomTest {
	int page = 1;
	NFTClient client = new NFTClient(key, secret);

	@Override
	protected NFTClient getClient() {
		return client;
	}

	@Test
	void testGetAssets() throws ApiException {
		testNotThrow(getClient().getAssets());
	}

	@Test
	void testGetAssets2() throws ApiException {
		testNotThrow(getClient().getAssets(new Paging(1)));
	}

	@Test
	void testGetDeposits() throws ApiException {
		testNotThrow(getClient().getDeposits());
	}

	@Test
	void testGetDeposits2() throws ApiException {
		testNotThrow(getClient().getDeposits(new FramedPaging(page)));
	}

	@Test
	void testGetTransactions() throws ApiException {
		for (var o : OrderType.values()) {
			testNotThrow(getClient().getTransactions(new TransactionHistoryParams(o)));
		}
	}

	@Test
	void testGetTransactions2() throws ApiException {
		testNotThrow(
				getClient().getTransactions(new TransactionHistoryParams(OrderType.SELL_ORDER),
						new FramedPaging(page)));
	}

	@Test
	void testGetWithdraws() throws ApiException {
		testNotThrow(getClient().getWithdraws());
	}

	@Test
	void testGetWithdraws2() throws ApiException {
		testNotThrow(getClient().getWithdraws(new FramedPaging(1)));
	}
}

File: src/test/com/binance4j/connectors/nft/client/NFTClientTest.kt
package com.binance4j.connectors.nft.client

import com.binance4j.connectors.connectors.Connectors
import com.binance4j.connectors.core.test.CustomTest
import com.binance4j.connectors.nft.dto.OrderType
import org.junit.jupiter.api.Test

internal class NFTClientTest : CustomTest() {
    private var client = Connectors.rest(key, secret).nft

    @Test
    fun `Get all transactions`() = OrderType.values().forEach { type -> assertDoesNotThrow(client.getTransactions(type)) }

    @Test
    fun `Get all transactions in an interval`() = OrderType.values().forEach { type ->
        assertDoesNotThrow(client.getTransactions(type, lastYear, today, 1, limit))
    }

    @Test
    fun `Get deposits in interval`() = assertDoesNotThrow(client.getDeposits(lastYear, today, 1, limit))

    @Test
    fun `Get withdraws in interval`() = assertDoesNotThrow(client.getWithdraws(lastYear, today, 1, limit))

    @Test
    fun `Get assets in interval`() = assertDoesNotThrow(client.getAssets(1, limit))
}